renderdoc/core/intervals_tests.cpp

```diff
@@ -0,0 +1,417 @@
+/******************************************************************************
+* The MIT License (MIT)
+*
+* Copyright (c) 2019 Baldur Karlsson
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+******************************************************************************/
+
+#include "intervals.h"
+#include "common/globalconfig.h"
+
+#if ENABLED(ENABLE_UNIT_TESTS)
+
+#include "3rdparty/catch/catch.hpp"
+
+#include <stdint.h>
+#include <vector>
+
+struct Interval
+{
+  uint64_t start;
+  uint64_t value;
+  uint64_t end;
+};
+
+std::vector<Interval> mkIntervals(std::vector<uint64_t> &vals)
+{
+  Interval interval = Interval();
+  std::vector<Interval> intervals;
+  for(auto i = vals.begin(); i != vals.end(); i++)
+  {
+    interval.value = *i;
+    i++;
+    if(i == vals.end())
+      interval.end = UINT64_MAX;
+    else
+      interval.end = *i;
+    intervals.push_back(interval);
+    interval.start = interval.end;
+  }
+  if(interval.end != UINT64_MAX)
+  {
+    interval.value = 0;
+    interval.end = UINT64_MAX;
+    intervals.push_back(interval);
+  }
+  return intervals;
+}
+
+void check_intervals(Intervals<uint64_t> &value, std::vector<Interval> expected)
```

Comment:
Nit:
```suggestion
void check_intervals(Intervals<uint64_t> &value, const std::vector<Interval> &expected)
```

It doesn't really matter for test code but it's good to keep things consistent and pass by ref when we don't need a copy. There are a couple of other helper functions here taking `vector`s by value that could be changed.

Reply:
Done